addons/payment/static/src/interactions/payment_form.js

```diff
@@ -20,7 +20,7 @@ export class PaymentForm extends Interaction {
     setup() {
         // Load the payment context from the payment form dataset.
         this.paymentContext = {};
-        Object.assign(this.paymentContext, this.el.dataset);
+        Object.assign(this.paymentContext, this.el.querySelector("#o_payment_form").dataset);
```

Comment:
This raises error when there are no payment methods available
<img width="1905" height="823" alt="image" src="https://github.com/user-attachments/assets/bf1233ea-e49e-49e9-a145-f3b04793326f" />